### PR TITLE
SOLR-15010 Try to use jattach for threaddump if jstack is missing

### PR DESCRIPTION
**Pull request by epugh**
Introduce jattach check if jstack is missing.  jattach ships in the Solr docker image instead of jstack.

# Description

With the Solr Docker image you get an alarming message about jstack being missing, and meanwhile we have jattach installed in our Docker image, so lets use that if we need to kill Solr.

# Solution

Checks if jattach exits, and adds it as a command, and then if jstack is missing, runs the jattach threaddump command.

# Tests

Hard to test!   I tweaked the bin/solr command to not actually shut solr down, in order to trigger the thread dump, then tested by building the Solr 9 Docker image, and issuing to the image called `solr3`:
```
docker exec solr3 solr stop -p 8983
Sending stop command to Solr running on port 8983 ... waiting up to 180 seconds to allow Jetty process 16 to stop gracefully.
Solr process 16 is still running; jattach threaddumping it now.
Connected to remote JVM
Response code = 0
2021-01-08 22:01:34
Full thread dump OpenJDK 64-Bit Server VM (11.0.9.1+1 mixed mode):
```

I then tested it with jstack by running it locally, but with the same commented out code and it worked, and then finally restored the commented out code and the start/stop cycle worked.

I did NOT change anything for Windows.

# Checklist

Please review the following and check all that apply:

- [ X] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [ X] I have created a Jira issue and added the issue ID to my pull request title.
- [ X] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [ X] I have developed this patch against the `master` branch.
- [ X] I have run `./gradlew check`.
- [ ] I have added tests for my changes.
- [ ] I have added documentation for the [Ref Guide](https://github.com/apache/lucene-solr/tree/master/solr/solr-ref-guide) (for Solr changes only).

### CODE REVIEW

**Review by HoustonPutman — APPROVED** [review]
Awesome, this looks great.

Thanks for taking the initiative on this, people have been complaining for a while.

**Review by janhoy — APPROVED** [review]
LGTM. Thanks for following up

**Comment by dsmiley** [comment]
Thanks Eric!

File: solr/bin/solr [hunk 1]
```diff
@@ -110,9 +110,10 @@ elif [ -n "$JAVA_HOME" ]; then
       JAVA="$java/java"
       if [ -x "$java/jstack" ]; then
         JSTACK="$java/jstack"
+      elif [ -x "$(command -v jattach)" ]; then
+        JATTACH="jattach"
```
Comment (cpoerschke) [inline solr/bin/solr:114]: minor/maybe suggestion:
```suggestion
        JATTACH="$(command -v jattach)"
```

**Pull request merged**